### DISCUSSION TITLE
Using CSS grid templates with vanilla extract

### DISCUSSION THREAD

**Discussion by oddnavy** [vanilla-extract-css/vanilla-extract#798] [Q&A]
Has anyone managed to add support for CSS grid templates in combination with vanilla extract and sprinkles?

I'm hoping for something 👇🏻  where any valid value for the CSS property grid-template-columns is allowed (where the `gridTemplateColumns` prop is passed through to `sprinkles()`):

```jsx
<Box display="grid" gridTemplateColumns="1fr 1fr" />
<Box display="grid" gridTemplateColumns="repeat(12, 1fr)" />
<Box display="grid" gridTemplateColumns="200px repeat(auto-fill, 100px) 300px" />
```

I could allow specific values but wondering if there is a 'pass-through' type option

**Comment by askoufis** [comment]
[Rainbow Sprinkles](https://github.com/wayfair/rainbow-sprinkles) might be what you're looking for.

**Comment by brandonpittman** [comment]
DessertBox has an escape hatch where you could add `__gridTemplateColumns="xxx"` as well. You could also just use `style="grid-template-columns: xxx"` when necessary.
